tapir/coop/services/membership_resignation_service.py

```diff
@@ -97,15 +97,16 @@ def on_resignation_deleted(cls, resignation: MembershipResignation):
 
     @staticmethod
     def delete_end_dates(resignation: MembershipResignation):
-        resignation.share_owner.share_ownerships.filter(
-            end_date=resignation.cancellation_date
-        ).update(end_date=None)
+        for share_ownership in resignation.share_owner.share_ownerships.all():
+            if share_ownership.end_date >= resignation.cancellation_date:
+                share_ownership.end_date = None
+                share_ownership.save()
```

Comment:
I think we want to be more specific about which shares end_date we remove. For the cases where the cancellation was instant, we can remove the end date of all shares that are after the cancellation date, as you did.

For `ResignationType.BUY_BACK`, we should only remove the end dates of shares that are at or after the pay out day.

When updating several shares, you can do it this way:
```suggestion
        resignation.share_owner.share_ownerships.filter(
            end_date__gte=resignation.cancellation_date
        ).update(end_date=None)
```